.github/workflows/push_pr.yml

```diff
@@ -30,6 +30,8 @@ jobs:
         include:
           - flink: 1.18.1
             jdk: '8, 11, 17'
+          - flink: 1.19-SNAPSHOT
```

Comment:
Is this the only relevant thing in this PR? Everything else looks like refactoring...

After https://github.com/apache/flink/pull/24249 was merged, this compiles which confirms no breaking changes?

Reply:
Are the rest of the changes due to removed internal APIs from Flink (e.g. the metrics stuff)? I would rename this PR as "Add build option for 1.19 to verify SinkV2 backward compatibility" or similar